isis/tests/AngleTest.cpp

```diff
@@ -0,0 +1,304 @@
+#include "Angle.h" 
+#include "SpecialPixel.h"
+#include "Constants.h" 
+#include "IException.h"
+
+#include <QString>
+#include <QDebug> 
+
+#include <gtest/gtest.h>
+
+#define TEST_EPSILON 1e-8
+
+TEST(AngleTest, DefaultConstructor) {
+  Isis::Angle angle;
+  EXPECT_FALSE(angle.isValid());
+  EXPECT_EQ(angle.radians(), Isis::Null);
+  EXPECT_EQ(angle.degrees(), Isis::Null); 
+  EXPECT_STREQ(angle.toString().toLatin1(), ""); 
+}
+
+
+TEST(AngleTest, NullInput_DegreeOutput) {
+  Isis::Angle angle;
+  EXPECT_NEAR(angle.degrees(), Isis::Null, TEST_EPSILON);
+  EXPECT_FALSE(angle.isValid());
+  EXPECT_STREQ(angle.toString().toLatin1(), ""); 
+}
+
+
+TEST(AngleTest, DegreeInput_RadianOutput) {
+  Isis::Angle angle(30., Isis::Angle::Degrees );
+  EXPECT_NEAR(angle.radians(), 30 * Isis::PI/180.0, TEST_EPSILON);
+  EXPECT_TRUE(angle.isValid()); 
+  EXPECT_STREQ(angle.toString().toLatin1(), "30.0 degrees");  
+}
+
+
+TEST(AngleTest, RadianInput_DegreeOutput) {
+  Isis::Angle angle(30. * Isis::PI / 180.0, Isis::Angle::Radians );
+  EXPECT_NEAR(angle.degrees(), 30.0, TEST_EPSILON);
+  EXPECT_STREQ(angle.toString(false).toLatin1(), "30.0");  
+}
+
+
+TEST(AngleTest, CopyConstructor) {
+  Isis::Angle angle(30., Isis::Angle::Degrees );
+  Isis::Angle angleCopy(angle);
+  EXPECT_NEAR(angleCopy.degrees(), 30.0,TEST_EPSILON); 
+}
+
+
+TEST(AngleTest, Mutators) {
+  Isis::Angle angle(30.0, Isis::Angle::Degrees );
+  angle.setDegrees(180);
+  EXPECT_NEAR(angle.degrees(), 180.0, TEST_EPSILON);
+  angle.setRadians(Isis::PI);
+  EXPECT_NEAR(angle.radians(), Isis::PI, TEST_EPSILON); 
+}
+
+TEST(AngleOperators, QDebug) {
+  // Stream qDebug output to a string and then do a string comparison?
+  Isis::Angle angle(45.0, Isis::Angle::Degrees);
+  QString qdebug_output;
+  QDebug debug(&qdebug_output);
+  debug << angle;
+  EXPECT_STREQ(qdebug_output.toLatin1(), "0.785398 <radians> (45 <degrees>) "); 
```

Comment:
Comparisons of strings with decimal places makes me feel un-easy. Could this be tested by just setting a 1 radian angle and outputing it?

Reply:
:+1:  to the idea of doing a fixed. no FP tests. As a follow on, are these debugs something that is valuable to test?

Reply:
@jlaura They are used for some file outputs, so it's worth testing that they properly output the angle as a string.

Reply:
@jessemapel One complication with the "change it to 1 radian" here is that the angle's value is output in both radians and degrees. I don't think I can avoid something with decimal places unless I use 0.

Reply:
(currently updating this to use 0)

Reply:
Sounds good :+1: